### ISSUE TITLE
Inlay hint is not removed when the line is changed

### ISSUE THREAD

**Issue by 0xd4d** [rust-analyzer/rust-analyzer#3056]
Version: 3e1d97790be166f8735607c552a94a28ab9b09b8

- Add this code
```rust
	let _ = 1u64.wrapping_add(1);
```
- Save it to get the inlay param hint
- This is what it looks like in VSCode before I press enter, `|` is the caret. Notice the `rhs: ` that was added by rust-analyzer.
```rust
	|let _ = 1u64.wrapping_add(rhs: 1);
```
- Press enter.
- What I see in VSCode, `|` is the caret:
```rust
	rhs:
	|let _ = 1u64.wrapping_add(1);
```

It disappears when I save the document.

**Comment by bjorn3** [comment]
This is similar to #3008.

**Comment by flodiebold** [comment]
Might actually be the same issue, if this is using the `onEnter` command...

**Comment by Veetaha** [comment]
I did have these bugs with inlay hints, but they are very random and seem to be a VSCode rendering bug. Though I am not sure, this needs more thorough investigation anyhow...

Kudos for the stable reproduction, @0xd4d!

**Comment by max-frai** [comment]
It's impossible to use rust-analyzer now due to this.
Everything is messing up every few seconds. Inline hints, highlighting, everything is misspositioned and wrong highlighted.

**Comment by 0xd4d** [comment]
@max-frai You can disable this in the settings until this issue is fixed. You can also add a thumbs up at the top and maybe they'll prioritize issues with most up votes.

**Comment by max-frai** [comment]
@0xd4d yes, thank you. Now I disabled inline hints and custom color highlighting. But it was very useful for me :(

**Comment by Veetaha** [comment]
@max-frai, may I ask you to provide a screenshot where the highlighting is broken, please. I remember having this kind of bug, but I am not sure that the bad highlighting behavior is the same as on your PC...

**Comment by Veetaha** [comment]
I also do think we need to prioritize bug-fixing on vscode client side. @matklad would we want to prioritize the implementation of  the command to update prebuilt LSP binary #3073, untill the next release, or we should we better tackle bugs?

**Comment by Veetaha** [comment]
@max-frai, @0xd4d, shame on me guys, I fixed my mess in #3173
